cmd/sovereignnode/chainSimulator/tests/bridge/bridge_test.go

```diff
@@ -0,0 +1,221 @@
+		"@000000000000000005004c13819a7f26de997e7c6720a6efe2d4b85c0609c9ad" + // price_aggregator_address
+		"@" + hex.EncodeToString([]byte("USDC-350c4e")) + // usdc_token_id
+		"@" + hex.EncodeToString([]byte("WEGLD-a28c59")) // wegld_token_id
+	feeMarketAddress := chainSimulatorIntegrationTests.DeployContract(t, cs, wallet.Bytes, &nonce, systemScAddress, feeMarketArgs, feeMarketWasmPath)
+
+	setFeeMarketAddressData := "setFeeMarketAddress" +
+		"@" + hex.EncodeToString(feeMarketAddress)
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, esdtSafeAddress, big.NewInt(0), setFeeMarketAddressData, uint64(10000000))
+
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, feeMarketAddress, big.NewInt(0), "disableFee", uint64(10000000))
+
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, esdtSafeAddress, big.NewInt(0), "unpause", uint64(10000000))
+
+	multiSigAddress := chainSimulatorIntegrationTests.DeployContract(t, cs, wallet.Bytes, &nonce, systemScAddress, "", multiSigWasmPath)
+
+	setEsdtSafeAddressData := "setEsdtSafeAddress" +
+		"@" + hex.EncodeToString(esdtSafeAddress)
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, multiSigAddress, big.NewInt(0), setEsdtSafeAddressData, uint64(10000000))
+
+	setMultiSigAddressData := "setMultisigAddress" +
+		"@" + hex.EncodeToString(multiSigAddress)
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, esdtSafeAddress, big.NewInt(0), setMultiSigAddressData, uint64(10000000))
+
+	setSovereignBridgeAddressData := "setSovereignBridgeAddress" +
+		"@" + hex.EncodeToString(multiSigAddress)
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, esdtSafeAddress, big.NewInt(0), setSovereignBridgeAddressData, uint64(10000000))
+
+	supply, _ := big.NewInt(0).SetString("123000000000000000000", 10)
+	issueArgs := "issue" +
+		"@" + hex.EncodeToString([]byte("MainToken")) +
+		"@" + hex.EncodeToString([]byte("TKN")) +
+		"@" + hex.EncodeToString(supply.Bytes()) +
+		"@" + fmt.Sprintf("%X", 18) // num decimals
+	issueTx := chainSimulatorIntegrationTests.SendTransactionAndWaitForMaxNrOfBlocks(t, cs, wallet.Bytes, &nonce, systemEsdtScAddress, issueCost, issueArgs, uint64(60000000), 3) // 3 blocks to execute on destination shard (meta)
+	tokenIdentifier := issueTx.Logs.Events[0].Topics[0]
+	require.True(t, len(tokenIdentifier) > 7)
+
+	err = cs.GenerateBlocks(1) // one more block to get the tokens in source shard
+	require.Nil(t, err)
+
+	tokens, _, err := cs.GetNodeHandler(0).GetFacadeHandler().GetAllESDTTokens(wallet.Bech32, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.NotNil(t, tokens)
+	require.True(t, len(tokens) == 1)
+	require.Equal(t, supply.String(), tokens[string(tokenIdentifier)].GetValue().String())
+
+	amountToDeposit, _ := big.NewInt(0).SetString("2000000000000000000", 10)
+	depositArgs := "MultiESDTNFTTransfer" +
```

Comment:
This test is very weird.
You issue a token in main chain, then deposit the exact token on your account and surprisingly you are minting "infinite" more money. This test will most probably fail in newer updates. We should have it in rc, where we also have cross chain actions + esdt prefix